File: javascript/decorator/StarbuzzCoffee/decorator/condiments/Milk.js
import {CondimentDecorator} from "../CondimentDecorator.js";
import {Beverage} from "../../component/Beverage.js";

export class Milk extends CondimentDecorator {

    beverage = new Beverage()

    constructor(beverage) {
        super();
        this.beverage = beverage
    }

    getDescription() {
        return this.beverage.getDescription() + ", Milk"
    }

    cost() {
        return this.beverage.cost() + 0.89
    }

}

File: cpp/decorator/StarbuzzCoffee/decorator/condiments/Milk.cpp
#include "../CondimentDecorator.h"

class Milk : public CondimentDecorator {

public:
    explicit Milk(Beverage* beverage) { this->beverage = beverage; }

    string getDescription() override { return beverage->getDescription() + ", Milk"; }

    double cost() override { return beverage->cost() + 0.89; }

private:
    Beverage* beverage;

};

